src/db/postgres/lrsql/postgres/sql/ddl.sql

```diff
@@ -280,3 +280,40 @@ CREATE TYPE scope_enum AS ENUM (
   'profile',
   'profile/read');
 ALTER TABLE IF EXISTS credential_to_scope ALTER COLUMN scope TYPE scope_enum USING (scope::scope_enum);
+
+
+/* Migration 2022-05-08-00 - Add timestamp to xapi_statement */
+
+-- :name query-xapi-statement-timestamp-exists
+-- :command :query
+-- :result :one
+-- :doc Query to see if `xapi_statement.timestamp` exists.
+SELECT 1 FROM information_schema.columns WHERE table_name = 'xapi_statement' AND column_name = 'timestamp';
+
+-- :name alter-xapi-statement-add-timestamp!
+-- :command :execute
+-- :doc Add `xapi_statement.timestamp` to allow easier timestamp access.
+ALTER TABLE xapi_statement ADD COLUMN timestamp TIMESTAMPTZ
+
+-- :name migrate-xapi-statement-timestamps!
+-- :command :execute
+-- :doc Backfill `xapi_statement.timestamp` with the values from the payload
+UPDATE xapi_statement SET timestamp = (payload->>'timestamp')::timestamptz WHERE timestamp IS NULL;
```

Comment:
Why the separate update functions? If you look at `alter-scope-enum-type!`, you'll see that multiple SQL updates are being performed in the same function.

Reply:
Yeah, in general im a big fan of conditional transactions where available but in this case because of the architecture I like the isolation and the parity with sqlite commands. Since it's not something that runs all the time, really only once per instance, it doesn't affect performance or anything so i kinda dig it.

Reply:
> and the parity with sqlite commands

That makes sense, but I would like to point out that this ship had already sailed a long time ago with the OIDC issuer and passhash updates.

Reply:
Yeah that's fine I'm not aiming to reinvent the wheel i just liked it in this case